en/concept/architecture.md

```diff
@@ -113,34 +113,27 @@ considerably slower.
 The message update rates can be [inspected](../middleware/uorb.md)
 in real-time on the system by running `uorb top`.
 
-## Runtime Environment
+## Runtime Environment {#runtime-environment}
 
-PX4 runs on various operating systems that provide a POSIX-API
-(such as Linux, macOS, NuttX or QuRT). It should also have some form of
-real-time scheduling (e.g. FIFO).
+PX4 runs on various operating systems that provide a POSIX-API (such as Linux, macOS, NuttX or QuRT).
+It should also have some form of real-time scheduling (e.g. FIFO).
 
-The inter-module communication (using [uORB](../middleware/uorb.md)) is based on shared memory. 
-The whole PX4 middleware runs in a single address space, i.e. memory is shared between all modules. 
+The inter-module communication (using [uORB](../middleware/uorb.md)) is based on shared memory.
+The whole PX4 middleware runs in a single address space, i.e. memory is shared between all modules.
 
-> **Info** The system is designed such that with minimal effort it would
-> be possible to run each module in separate address space (parts that would need
-> to be changed include `uORB`, `parameter interface`, `dataman` and `perf`).
+> **Info** The system is designed such that with minimal effort it would be possible to run each module in separate address space (parts that would need to be changed include `uORB`, `parameter interface`, `dataman` and `perf`).
 
 There are 2 different ways that a module can be executed:
-- **Tasks**: The module runs in its own task with its own stack and process priority
-  (this is the more common way). 
-- **Work queues**: The module runs on a shared task, meaning that it does not own a stack. 
-  Multiple tasks run on the same stack with a single priority per work queue.
+- **Tasks**: The module runs in its own task with its own stack and process priority.
+- **Work queues**: The module runs on a shared task, meaning that it does not own a stack.
+  Multiple modules can run on the same stack with a single priority per work queue.
```

Comment:
YOu say "same stack with a single priority per work queue." - do you mean "all work queues share a stack and priority?"

Reply:
No, there are N work queues, each of which has a priority + stack, and on each work queue there can be an arbitrary number of tasks/modules/work queue items running (a task needs to be assigned to a specific work queue).